File: 412. Fizz Buzz.cpp
class Solution {
public:
    // brute force, O(N), 59.53%, ok
    vector<string> fizzBuzz(int n) {
        vector<string> ans;
        for(int i = 1; i <= n; i++){
            if(i % 3 == 0 && i % 5 == 0)
                ans.push_back("FizzBuzz");
            else if(i % 3 == 0) ans.push_back("Fizz");
            else if(i % 5 == 0) ans.push_back("Buzz");
            else ans.push_back(to_string(i));
        }
        return ans;
    }

    // another way，O(N), 59.53%, ok
    vector<string> fizzBuzz(int n) {
        vector<string> ans(n, "");
        for(int i = 1; i <= n; i++){
            if(i % 3 == 0) ans[i - 1] += "Fizz";
            if(i % 5 == 0) ans[i - 1] += "Buzz";
            if(ans[i - 1] == "") ans[i - 1] = to_string(i);
        }
        return ans;
    }

};

File: 412. Fizz Buzz.py
class Solution(object):
    
    # Naive: O(N) time & space, 81.42%
    def fizzBuzz(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        ret = list()
        for i in xrange(1, n + 1):
            if i % 15 == 0:
                ret.append("FizzBuzz")
            elif i % 5 == 0:
                ret.append("Buzz")
            elif i % 3 == 0:
                ret.append("Fizz")
            else:
                ret.append(str(i))
        return ret
    
    # List Comprehension: O(N) time & space, 58.17%
    def fizzBuzz(self, n):
        return [('Fizz' * (not i % 3) + 'Buzz' * (not i % 5)) or str(i) for i in xrange(1, n + 1)]

